### PR TITLE
Huge refactoring of training scripts and new features to stabilize pitch and durations

### PR DESCRIPTION
**Pull request by r9y9**
I'm sorry for this quite a big PR, but here come many improvements, including new features. Most new features are inspired by the recent Sinsy's paper: https://arxiv.org/abs/2108.02776 (big thanks to the authors). Dev enhancements are inspired by my recent publication for text-to-speech https://github.com/r9y9/ttslearn.

### New sample (2022/03/12)

- [2022/03/12 NIT-SONG070 pitch regularization + MDN duration generation Nitech Jp Song070 F001 003](https://soundcloud.com/r9y9/20220311-nit-song070-pitch-regularization-mdn-duration-generation-nitech-jp-song070-f001-003?in=r9y9/sets/dnn-based-singing-voice)

### Previous sample (2020/11/16)

- [2020/11/16 NIT-SONG070 svs-world-conv-mdn dim-wise MDN Nitech Jp Song070 F001 003](https://soundcloud.com/r9y9/20201116-nit-song070-svs-world-conv-mdn-dim-wise-mdn-nitech-jp-song070-f001-003?in=r9y9/sets/dnn-based-singing-voice)

## New features

- Residual F0 prediction inside an acoustic model #76. 
  - This enables to generate more smooth and natural F0 trajectory.
- Pitch regularization loss for training acoustic models. 
  - This enables the model to generate more stable pitch that is close to the one in the musical score.
- Improved duration generation considering the variances #80. 
  - more natural durations can be obtained.

## Improvements

- Tenorboard logging is supported
- Modularized training scripts
- Fixed deprecations for hydra 1.1
- Multi-GPU training is supported 
- New training script (`nnsvs-train-resf0`) is added to train a new acoustic model

## Recipes

```
egs/nit-song070/svs-world-resf0
```

This recipe should reproduce the sample: https://soundcloud.com/r9y9/20220311-nit-song070-pitch-regularization-mdn-duration-generation-nitech-jp-song070-f001-003

### CODE REVIEW

**Comment by r9y9** [comment]
https://github.com/r9y9/nnsvs/pull/81/commits/4044f48dfbe04cb270fe5ed34769893ce05f35c0

I implemented new MDN-based residual F0 predition model. So far I couldn't make it work well. It seems we need much more iterations (>200 epochs). Non-MDN-based models perform well with 50 epochs though. RNN-based models may require a bit more iterations in my experience.

**Comment by r9y9** [comment]
For those who are watching the repo's activities: I may introduce some breaking changes during the refactoring process. If you encounter any issues, please let me know.  I will address concerns as much as I can.
